### DISCUSSION TITLE
Error handling during AST `transform`

### DISCUSSION THREAD

**Discussion by ShayRubach** [swc-project/swc#3409] [Q&A]
I would like to return some errors in my code which implements some `visit_mut`s. I know I know, they do not return anything, but I would like to append errors along the way (during transformation) and return them back to the client who called the `transform`ation.

I examined the `Handler` but I didn't quite figure out what do I do with it. Is that the direction to take? Or, alternatively are there any other solutions to it as far as you know?

Thanks! 🙇‍♂️

**Comment by pictographer** [comment]
I'm planning to do something like this for analytics. Seems like creating a shared BTreeMap (or whatever data structure you need) with Lazy Static or Once Cell should work.

**Comment by pictographer** [comment]
```rust
use once_cell::sync::Lazy;
use std::collections::BTreeSet;
use std::sync::Mutex;

static ERR_SET: Lazy<Mutex<BTreeSet<String>>> = Lazy::new(|| Mutex::new(BTreeSet::new()));

fn insert_err(err: &str) {
    ERR_SET.lock().unwrap().insert(err.to_string();
}
```
Slightly simplified from my code to collect function names. You'll probably want something better than the `unwrap()` and `String` error message type, etc.

I was delighted at how good the [Once Cell documentation](https://docs.rs/once_cell/latest/once_cell/) and API are.
